nuc_morph_analysis/analyses/density/figure_watershed_based_density_schematic.py

```diff
@@ -0,0 +1,229 @@
+#%%
+# code for SuppFigS4 panel C
+from nuc_morph_analysis.lib.preprocessing.twoD_zMIP_area import watershed_workflow, pseudo_cell_helper, pseudo_cell_testing_helper
+from pathlib import Path
+import pandas as pd
+from nuc_morph_analysis.lib.preprocessing import global_dataset_filtering
+from nuc_morph_analysis.lib.preprocessing import filter_data, load_data
+
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.cm as cm
+from nuc_morph_analysis.lib.visualization.plotting_tools import colorize_image, get_plot_labels_for_metric
+from nuc_morph_analysis.lib.visualization.notebook_tools import save_and_show_plot
+
+from nuc_morph_analysis.analyses.dataset_images_for_figures.figure_helper import return_glasbey_on_dark
+
+
+from nuc_morph_analysis.analyses.density.visually_validate_watershed_psuedo_cell_seg_workflow import get_contours_from_pair_of_2d_seg_image, draw_contours_on_image
+from nuc_morph_analysis.analyses.dataset_images_for_figures.figure_helper import INTENSITIES_DICT
+
+import matplotlib
+matplotlib.rcParams['pdf.fonttype'] = 42
+matplotlib.rcParams['ps.fonttype'] = 42
+matplotlib.rcParams['font.size'] = 7
+
+def determine_crop_size_and_location_from_track(track,df,crop_w=250,crop_h=250,RESOLUTION_LEVEL=1):
+    """
+    determine the crop size and location based on the track_id
+    """
+    crop_w = 250
+    crop_h = 250
+    track_df = df[(df['track_id']==track)]
+    track_x = track_df['centroid_x'].values[0]
+    track_y = track_df['centroid_y'].values[0]
+    if RESOLUTION_LEVEL ==1:
+        track_x = np.uint16(track_x//2.5)
+        track_y = np.uint16(track_y//2.5)
+
+    track_x = np.uint16(track_x - crop_w//2)
+    track_y = np.uint16(track_y - crop_h//2)
+    return track_x,track_y,crop_w,crop_h
+
+def determine_colormaps(img,key,crop_exp):
+    # determine colormaps and vmin,vmax
+    if img.dtype == 'bool':
+        cmap = cm.get_cmap('Greys_r')
+        vmin = 0
+        vmax = 1
+    elif img.dtype == 'uint16':
+        _, cmap, _ = return_glasbey_on_dark(N=img.max()+1,from_list=True)
+        vmax = cmap.N-1
+        vmin = 0
+    else:
+        cmap = cm.get_cmap('Greys_r')
+        vmin=0
+        vmax=np.max(img[crop_exp])
+
+    if key=='raw_image':
+        cmap = cm.get_cmap('Greys_r')
+        vmin = INTENSITIES_DICT['egfp_max'][0]
+        vmax= INTENSITIES_DICT['egfp_max'][1]
+    return cmap, vmin, vmax
+
+
+def run_validation_and_plot(TIMEPOINT=88,track=81463,colony='medium',RESOLUTION_LEVEL=1,plot_everything=False, testing=False):
```

Comment:
can this be added and pulled from the example track dictionary? `nuc_morph_analysis/lib/visualization/example_tracks.py`